### PR TITLE
Demo environment

### PR DESCRIPTION
**Pull request by Omri-Levy**
### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

___

### **PR Type**
enhancement, bug_fix

___

### **Description**
- Enhanced the NoAlerts component by adjusting its bottom padding for better layout.
- Improved error logging in WinstonLogger by including payload data.
- Added new Prisma transaction types to enhance database transaction handling.
- Fixed a typo in a TypeScript comment to improve code clarity.
- Updated project dependencies in `pnpm-lock.yaml` and `package.json` to include new packages and adjust TypeScript versions.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>NoAlerts.tsx</strong><dd><code>Adjust bottom padding in NoAlerts component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/NoAlerts/NoAlerts.tsx
- Reduced the bottom padding of the NoAlerts component.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2384/files#diff-31ab3888714632cf6d87e58c8a254ae90b9cbc1fae371e73a47c2776feb454f2">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>winston-logger.ts</strong><dd><code>Enhance error logging with payload support</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/src/common/utils/winston-logger/winston-logger.ts
<li>Refactored the error logging to include a payload.<br> <li> Minor import reordering.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2384/files#diff-fd4a7890299f448e018f3d76daa70a5c81881da70ed7e1b8da421591f5e23e1d">+6/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>types.ts</strong><dd><code>Define new Prisma transaction types</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/src/types.ts
- Added new types for Prisma transaction handling.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2384/files#diff-5c2546c66fed2c9ea8b17e9691a6ca83caa45ed68e4a369c41844dee2ea3765b">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pnpm-lock.yaml</strong><dd><code>Update dependencies and TypeScript version in lock file</code>&nbsp; &nbsp; </dd></summary>
<hr>

pnpm-lock.yaml
<li>Added dependencies for '@types/jmespath' and 'jmespath'.<br> <li> Updated TypeScript version in dependencies.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2384/files#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bb">+8/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Add new package dependencies</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/package.json
- Added new package dependencies for 'jmespath' and its types.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2384/files#diff-65672d584fae78ff93600b1898b6b95bb50b5e51a25297cd36a840c37cd34b65">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Bug_fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>hook-callback-handler.service.ts</strong><dd><code>Fix typo in TypeScript comment</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/src/workflow/hook-callback-handler.service.ts
- Corrected a typo in a TypeScript comment.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2384/files#diff-fdcf34223f85ff30a15d709b24b380b5d53c37e98cf3bcbfd255b581c919eca1">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

## Summary by CodeRabbit

- **Bug Fixes**
  - Adjusted padding in the "No Alerts" component to improve layout consistency.

- **New Features**
  - Added new dependencies: `@types/jmespath` and `jmespath` for enhanced functionality.

- **Improvements**
  - Enhanced error logging to include additional payload information for better debugging.
  - Added new types for Prisma transactions to streamline database operations.
  - Improved document update logic to ensure updates only occur when necessary.

- **Code Cleanup**
  - Removed unnecessary function calls in the tab-block mapping logic.
  - Minor comment update in the hook callback handler service.

### CODE REVIEW

File: services/workflows-service/src/types.ts [hunk 1]
```diff
@@ -40,3 +40,7 @@ export type GenericFunction = (...args: AnyArray) => any;
 export type GenericAsyncFunction = (...args: AnyArray) => Promise<any>;
 
 export type PrismaTransaction = Omit<PrismaClient, runtime.ITXClientDenyList>;
+
+export type PrismaTransactionMethod = Parameters<PrismaClient['$transaction']>[0];
+
+export type PrismaTransactionClient = Parameters<PrismaTransactionMethod>[0];
```
Comment (Omri-Levy) [inline services/workflows-service/src/types.ts:46]: Used in data-migrations PR.

File: services/workflows-service/package.json [hunk 2]
```diff
@@ -140,6 +141,7 @@
     "eslint-plugin-import": "^2.27.5",
     "jest": "29.7.0",
     "jest-mock-extended": "^2.0.4",
+    "jmespath": "^0.16.0",
```
Comment (Omri-Levy) [inline services/workflows-service/package.json:144]: Used in rule tests.

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/NoAlerts/NoAlerts.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { FunctionComponent } from 'react';
 
 export const NoAlerts: FunctionComponent = () => {
   return (
-    <div className="flex items-center justify-center p-4 pb-72">
+    <div className="flex items-center justify-center p-4 pb-64">
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/NoAlerts/NoAlerts.tsx:6]: Fixes overflow in alerts view.

File: apps/backoffice-v2/src/domains/workflows/hooks/mutations/useUpdateDocumentByIdMutation/useUpdateDocumentByIdMutation.tsx [hunk 4]
```diff
@@ -41,6 +41,10 @@ export const useUpdateDocumentByIdMutation = ({
       const previousWorkflow = queryClient.getQueryData(workflowById.queryKey);
 
       queryClient.setQueryData(workflowById.queryKey, (oldWorkflow: TWorkflowById) => {
+        if (!oldWorkflow) {
+          return;
+        }
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/domains/workflows/hooks/mutations/useUpdateDocumentByIdMutation/useUpdateDocumentByIdMutation.tsx:46]: Trying to update an associated company document would error due to workflow being undefined.

File: apps/backoffice-v2/src/lib/blocks/variants/DefaultBlocks/hooks/useCaseBlocksLogic/utils/get-tabs-block-map.tsx [hunk 5]
```diff
@@ -68,7 +68,6 @@ export const getTabsToBlocksMap = (
     associated_companies: [
       ...associatedCompaniesBlock,
       ...associatedCompaniesInformationBlock,
-      ...createKycBlocks(workflow as TWorkflowById),
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/variants/DefaultBlocks/hooks/useCaseBlocksLogic/utils/get-tabs-block-map.tsx:71]: Associated companies tab is not supposed to have KYC blocks.

**Review by MatanYadaev — APPROVED** [review]

**Pull request merged**